### ISSUE TITLE
Train a Neural Network to detect if an image is a tweet

### ISSUE THREAD

**Issue by giulionf** [giulionf/realtweetornotbot#57]

**Comment by Shivam60** [comment]
Any dataset or any idea how to create such a dataset?

**Comment by giulionf** [comment]
I have written Code to crawl the Dataset. I can Offer a package with 20000 Twitter Screenshots and 20000 normal "meme" Images as negatives. If needed more could ne crawled! Would that be sufficient?

**Comment by Shivam60** [comment]
I think that it is more than enough. Could you mention the image size and maybe show an example of each class?

**Comment by giulionf** [comment]
Image sizes are variable. The solution would require a downscale/variable input!

Examples for tweet class:
![3210733](https://user-images.githubusercontent.com/24531319/54493838-a4672300-48d4-11e9-8668-dd8f0a26ec94.png)
![183275398908481536](https://user-images.githubusercontent.com/24531319/54493840-a9c46d80-48d4-11e9-8c88-00bd44b8ff3e.png)

Examples of non tweet class:
![bad (11274)](https://user-images.githubusercontent.com/24531319/54493843-b34dd580-48d4-11e9-8d7e-560f4a804bab.jpg)
![bad (11334)](https://user-images.githubusercontent.com/24531319/54493850-bd6fd400-48d4-11e9-832c-7ae4b155fa83.jpg)

**Comment by Shivam60** [comment]
okay, I am interested in solving this. Can you share the dataset?

**Comment by giulionf** [comment]
Sure, I will upload the data and send it to you once it's done! Just remember that this is not an open dataset and can not be used for anything but this issue without my permission!

**Comment by Shivam60** [comment]
cool, got it. should I push my work in this repo?

**Comment by giulionf** [comment]
@Shivam60 I'm thinking about seperating it into a second repository with the rest of the code (Dataset generator). Or do you want to do it via a pull request? I think it would be cool to have it usable as a seperate python module that the bot includes!

**Comment by Shivam60** [comment]
Well trying a resnet didn't help. Any ideas?

**Comment by giulionf** [comment]
Do you have conv-layers in the net? I just thought about it... Maybe experimenting with the kernel size would yield better results? What exactly did you try?

I remember when I tried this I used keras with 2 conv. layers of the size 64 and 32... Maybe we should think about how we pre process the images... How did you deal with the variable input sizes in your approach?

**Comment by Shivam60** [comment]
Well, I Resnet50 requires 224 by 224 shape so I set the target size in my image generator for that. So all images were resized to it. I normalized the dataset but still didn't get much accuracy. I suspect that CNN look for patterns which repeat themselves and in these images there is none. Might we try to extract text from images and look for them in twitter?

**Comment by giulionf** [comment]
Thats what Im doing atm. I'm extracting text using tesseract, which serves as a bottleneck to the rest of the application. The bot runs on heroku which only allows 512 MB of Ram, so I had to put a lock onto the tesseract operation. The whole thing could be severly speed up by a NN that detects if the image is a tweet before doing OCR...

I had the same results as you did with my CNN approach... I guess my problem was that the aspect ratio of the tweet was not kept and thus the Convultional Layer could not get things like the circle of the Avatar or similar stuff....

**Comment by giulionf** [comment]
Issue closed since it seems too complex for what it‘s worth.
